src/output/fmt.rs

```diff
@@ -1,187 +1,120 @@
+use crate::util::termcolor::{Buffer, BufferWriter, ColorChoice};
 #[cfg(feature = "color")]
-use ansi_term::{
-    ANSIString,
-    Colour::{Green, Red, Yellow},
-};
-
-use std::env;
-use std::fmt;
-
-#[derive(Debug, Copy, Clone, PartialEq)]
-pub(crate) enum ColorWhen {
-    Auto,
-    Always,
-    Never,
-}
+use crate::util::termcolor::{Color, ColorSpec, WriteColor};
+
+use std::fmt::{self, Debug, Formatter};
+use std::io::{Result, Write};
 
 #[cfg(feature = "color")]
-pub(crate) fn is_a_tty(stderr: bool) -> bool {
+fn is_a_tty(stderr: bool) -> bool {
     debugln!("is_a_tty: stderr={:?}", stderr);
+
     let stream = if stderr {
         atty::Stream::Stderr
     } else {
         atty::Stream::Stdout
     };
+
     atty::is(stream)
 }
 
 #[cfg(not(feature = "color"))]
-pub(crate) fn is_a_tty(_: bool) -> bool {
+fn is_a_tty(_: bool) -> bool {
     debugln!("is_a_tty;");
     false
 }
 
-pub(crate) fn is_term_dumb() -> bool {
-    env::var("TERM").ok() == Some(String::from("dumb"))
-}
-
-pub(crate) struct ColorizerOption {
-    pub(crate) use_stderr: bool,
-    pub(crate) when: ColorWhen,
+pub struct Colorizer {
```

Comment:
They are gated by output. `mod output` in `src/lib.rs`. I don't understand why we need `pub(crate)` here.

Reply:
The benefit of `pub(crate)` is that the publicity statement is very apparent, very *visible*. From only a look at it I can immediately assume, "OK, this is not exposed to public, I can tweak this as much as I wish without breaking public API". 

When I see `pub`, there's no such certainty. Is this really public? Or is this `pub` just because someone was too lazy to type seven extra characters (no offense, I am a lazy butt too). To actually check it, I would have to step back and go look *elsewhere* (`lib.rs` in our case); yet another place (if only one!) to consider.

This is why I'm striving for "`pub` is only for really public stuff" policy.